File: src/main/java/com/akagiyui/common/exception/CustomException.java
package com.akagiyui.common.exception;

import com.akagiyui.common.ResponseEnum;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

/**
 * 自定义异常
 * @author AkagiYui
 */
@Getter
@Setter
@AllArgsConstructor
public class CustomException extends RuntimeException {
    /**
     * 响应状态枚举
     */
    private final ResponseEnum status;
}

File: src/main/kotlin/com/akagiyui/common/exception/CustomException.kt
package com.akagiyui.common.exception

import com.akagiyui.common.ResponseEnum


/**
 * 自定义异常
 *
 * @author AkagiYui
 */
class CustomException(
    /**
     * 响应状态枚举
     */
    private val status: ResponseEnum
) : RuntimeException() {
    fun getStatus() = status
}
